### DISCUSSION TITLE
[Auth] Independent lockscreen

### DISCUSSION THREAD

**Discussion by NikolaLozanovski** [ente-io/ente#1016] [Enhancements]
I have installed the desktop app on Ubuntu 23.10 from nightly build today, while setting the app I tried to enable the lockscreen option and popup was shown showing:

To enable lockscreen, please setup device passcode or screen lock in your system settings. As shown in the image added.

I would like to insert password when I enable this option. The android option to enable lockscreen works flawless, but works completely different.

![image](https://github.com/ente-io/auth/assets/16356700/717347e7-3fcb-4ffe-affd-8508fc23ff30)

**Comment by Damariobros** [comment]
This would be really nice as I do not want to have to use Windows Hello to secure the app! All using Windows Hello does is bring me endless notifications from the Settings app that there's a "problem with my microsoft account."

**Comment by Navet94** [comment]
It will be a good thing to have something like PIN or password to unlock the Windows App.

If someone access to my PC, I dont want they get my 2FA app too without something to block them.

**Comment by ZOMGsheikh** [comment]
> I have installed the desktop app on Ubuntu 23.10 from nightly build today, while setting the app I tried to enable the lockscreen option and popup was shown showing:
> 
> To enable lockscreen, please setup device passcode or screen lock in your system settings. As shown in the image added.
> 
> I would like to insert password when I enable this option. The android option to enable lockscreen works flawless, but works completely different.
> 
> ![image](https://private-user-images.githubusercontent.com/16356700/308945742-717347e7-3fcb-4ffe-affd-8508fc23ff30.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2ODkzNzksIm5iZiI6MTcyMjY4OTA3OSwicGF0aCI6Ii8xNjM1NjcwMC8zMDg5NDU3NDItNzE3MzQ3ZTctM2ZjYi00ZmZlLWFmZmQtODUwOGZjMjNmZjMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDEyNDQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmNDY2NzYyZWFjZTcxZjE3NTRhNmRkYTE3M2ZjZjc2OWE2ZDc1ZjA2YjVkMmNiNDEyNDljMDVhZmI5MGFmNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gcqw08fjkQP1-0CCoaNF2t18WF5Ezc7oE31LvpV2GBw)

Were you able to solve this? I am having similar issue over PopOS

**Reply by Damariobros** [reply]
Ente needs to add the functionality. Currently, on Windows the app lock only functions with Windows Hello PIN and Biometrics, on Mac, it only works with the device's on-board authentication, and on phones it only works with the device's lock screen PIN and biometrics. In other words, anyone who can access the device from a locked state can also access the app. People on Linux may or may not even be able to set up the lock screen at all.

So far, I believe Ente has added such functionality of a custom app lock to their main Ente app, but they still need to add it to Ente Auth.

**Comment by semyl** [comment]
I really want to see this too. It's great at least I can use my Windows PIN, but that is not secure enough - I may trust someone to use my PC but would not trust them to access my 2FA codes.

**Reply by nathfavour** [reply]
exactly

**Comment by sv3nnie** [comment]
This is now available in [v3.1.1](https://github.com/ente-io/ente/releases/tag/auth-v3.1.1) (see changelog of v3.1.0). You can enable it through settings -> Security -> App lock

**Reply by semyl** [reply]
Just installed v3.1.1 on Windows. App lock option is there but selecting "Pin lock" then does not let you type into the pin fields (nothing appears, highlighted box does not move), so pin cannot be set.

Password works though.

**Reply by Damariobros** [reply]
It's not working. It's just giving me the error, "No System Lock Found". It is still wanting me to set up and use Windows Hello PIN, there is no independent password or pin that I can enter into Ente to lock it.

Edit: After setting up Windows Hello PIN, I can enter Windows Hello PIN to access the lock screen settings, but I cannot enter anything into any of the fields; it is not letting me enter a new PIN or Password to switch over to them.

Edit 2: I'm not sure what happened, but after having Ente minimized for some time on the password enter screen, I can now type in the PIN and Password fields.

Edit 3: I'm messing around with the lock. It seems that, regardless of what the lock is set to, whether to the system lock, a PIN, or a password, I cannot access the app lock settings if Windows Hello is disabled. However disabling Windows Hello does not affect the existing app lock, and it continues to function correctly. 

Every time app lock is disabled, when you re-enable it, you will need to minimize and reopen the Ente window while on the screen to enter a new PIN or Password before you can start entering any PINs or passwords.

The lock screen always works correctly, it seems.

Also, btw, nice touch with allowing alphanumeric PINs, even special characters are allowed. That's cool, I like it.

Edit 4: And finally, it seems that all security settings are protected by the lock screen now... EXCEPT for messing around with your Ente account's passkeys, which is the most likely to lock you out of your account. Deleting your account should probably be protected as well. This needs to be fixed.


Now all that needs to happen is we need a lock now button so we don't have to fully close the app to lock it, as well as a lock timeout.


Edit 5: UM, I just found a SERIOUS problem! I don't know how I didn't notice this before! If you set the lock screen to a PIN or Password, and then disable Windows Hello, then ALL protected functions with the exception of the lock screen itself become unprotected! All functions such as changing your account password, viewing your recovery key, viewing changing your account email, viewing active sessions, etc, ALL of them become unprotected despite there being an app lock in place!

**Reply by enricodias** [reply]
Still doesn't work on linux (ubuntu). The app lock option just shows the same message "no system lock found".

**Reply by Damariobros** [reply]
Bump because I amended my above comment with a problem I found

**Reply by ashilkn** [reply]
@semyl @enricodias @Damariobros We're working on fixing these issue, thanks for reporting these!

**Reply by Damariobros** [reply]
When all this bug fixing is said and done, are there plans to port the custom pin and password setting to iOS and Android?

**Reply by ashilkn** [reply]
@Damariobros Yes, it's already done. Just have to roll out releases on both.

**Reply by Damariobros** [reply]
Just updated on my iPad! 👍

Super happy to have an independent lock on it now! It's starting to take shape as a suitable cross-platform replacement for Authy 👌

**Reply by nathfavour** [reply]
Ente is the best!

**Comment by Damariobros** [comment]
New update, and the biggest problems with the app lock as they relate to the system lock have been solved! 🥳

It seems the bug still remains where you have to minimize and reopen Ente when you create a new password or pin -- it seems to trigger whenever you use Windows Hello. For whatever reason you can't input anything to create a password or pin if you used Windows Hello PIN to access the App Lock settings.

In normal use, this bug will now only be encountered if you are switching from System Lock to one of the other two options.

Also, the Delete Account and Passkey options still need to be protected by the app lock, and we still need a time-out and a manual lock button, as well as the ability to hide the app to the system tray while it's locked.

**Reply by ashilkn** [reply]
Delete account is protected by the app lock (or device lock if the app lock isn't enabled). The app prompts for authentication on clicking the 'No, delete account' button inside the Delete account screen.

**Reply by Damariobros** [reply]
Oh good! That just leaves passkeys to protect. 👌

**Reply by ashilkn** [reply]
Had missed this, will protect passkeys in the next update :)

**Reply by Gobias64** [reply]
I agree that we also need a time-out lock setting and a manual lock. 1Password also has a setting to lock their app whenever my Mac is locked, which would be nice, too.

**Reply by Damariobros** [reply]
@Gobias64 Currently, those features are on the mobile version, albeit mutually exclusive --- you can set the timeout to "immediately" and then the home and sleep buttons become the lock buttons. The timeout needs to be ported to desktop, and then have a manual lock button added as well.

The ability to hide the app to the system tray while locked would be a nice change as well.

**Comment by NikolaLozanovski** [comment]
I've installed v3.1.3. on my desktop with Ubuntu 24.04 I can confirm there is an active password which protects the codes.
I am not sure if the password is synced between devices, if not it could be a nice feature, also prompt password again after inactivity.

**Reply by nathfavour** [reply]
Syncing password could reduce security greatly...I guess

**Reply by ashilkn** [reply]
The password or PIN for app lock in not synced across devices.

**Reply by nathfavour** [reply]
nice
